internal/controller/pipelinerollout/pipelinerollout_recycle.go

```diff
@@ -508,6 +584,9 @@ func calculateScaleForRecycle(
 				originalPodsRunning, found := historicalPodCount[vertexName]
 				if found {
 					newScaleValue = int64(math.Ceil(float64(originalPodsRunning) * float64(percent) / 100.0))
+					if newScaleValue < 1 {
+						newScaleValue = 1
+					}
```

Comment:
make sure that we don't allow the value to be < 1, which would prevent draining from working